projects/frontend/cicd/.gitlab-ci.yml

```diff
@@ -45,7 +45,7 @@ frontend-data-pipelines-e2e-tests:
   before_script:
     - cd projects/frontend/
   script:
-    - ./cicd/test_e2e_data_pipelines.sh "$VDK_API_TOKEN"
+    - ./cicd/test_e2e_data_pipelines.sh "$VDK_API_TOKEN" "$CYPRESS_test_environment" "$CYPRESS_grepTags"
```

Comment:
currently gitlab env variable is not set, but it could be if we want, using same setup for all projects

Reply:
Is this a draft PR?
Also, it seems that a lot of changes are put in this same Pr, please split it with a description of what is the change and why is it needed?

Reply:
I had allocated only half story point to stabilize e2e tests, because it's not in our sprint goals, but I decided to do it, because I could get stuck and not able to release new versions due to tests instability.
I already spend 1 and half points to make e2e tests work as expected, and I cannot spend more time to split this PR in smaller PR for every change here.
The only thing I could do it, is to discard all changes that are not related with e2e tests stabilization and not commit them anytime soon, because I don't have time for incremental improvements for every small change.
Schedules and estimation are pretty tight for SC, because we are already late and 3 weeks behind our main goals, due to backporting VDK UI and stabilization.
I think this will explain why I'm mixing multiple things in one PR.

Reply:
this are the changes included in this PR:
* Changes includes mainly propagated e2e tests from SC which were stabilized there, under different initiative (issue).
* With tests propagation there was need to add attributes on some HTML elements and minor changes in Components HTML templates, which shouldn't break anything and are validated manually and with e2e runs locally on my machine.
* There are minor scripts changes and added one env variable test_environment in gitlab variable to adopt backpropagated stabilized tests from SC.
* Improved sidebar menu navigation where Explore and Manage section are not expanded by default and works i sync with URL
* Added root spinner that would show loading state while static files are getting fetch in Browser before application get bootstrapped